internal/testrunner/system/system.go

```diff
@@ -0,0 +1,44 @@
+package system
+
+import (
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+)
+
+// Run runs the system tests for the given package.
+func Run(packageRootPath string) error {
+	systemTestsPath, err := findSystemTestsPath(packageRootPath)
```

Comment:
I'm thinking about the best place for filtering tests:

`elastic-package test --dataset ingest_controller`

The `--dataset` parameter filters test files based on the dataset criteria, which may be valid for both kinds of tests (pipeline, system). Do you think we can put this in some common place, before executing runners?

Reply:
I like this idea. Just to confirm, if no `--dataset` is specified we iterate over all datasets, yes?

> Do you think we can put this in some common place, before executing runners?

Yup, by specifying this flag as a persistent flag on the root `test` command it's value will be available to all sub-commands of `test`: `test system`, `test pipeline`, etc.

Reply:
Couple more questions about the `--dataset` flag:
- should we accept multiple occurrences of `--dataset` as a way for a user to specify multiple datasets?
- should we make it behave like the `-run` flag of `go test` in that it can take a regular expression and execute tests for matching datasets?

I'm also okay deferring one or both of the above features to follow up PRs. WDYT?

Reply:
> should we accept multiple occurrences of --dataset as a way for a user to specify multiple datasets?

I think the proposal defines multiple values for single flag: `--dataset=a,b,c`.

> should we make --dataset behave like the -run flag of go test in that it can take a regular expression and execute tests for matching datasets?

I admit I didn't think about it.

For flexibility we can add such feature, but I'm not sure about potential use cases. Usually dataset names differ much from each other. I don't reckon any names that have a common prefix. Another use case - run tests for all datasets except X.

Maybe just delay such feature until we have a real requirement.

> I'm also okay deferring one or both of the above features to follow up PRs. WDYT?

Absolutely right, it's not a first need feature.

Reply:
Implemented `--dataset` as an optional flag, allowing it to take a comma-separated values, in 165e30b.